src/software/jetson_nano/thunderloop.cpp

```diff
@@ -416,6 +417,29 @@ double Thunderloop::getCpuTemperature()
     }
 }
 
+bool isPowerStable(std::ifstream& log_file)
+{
+    // if the log file cannot be open, we would return false. Chances are, the battery
+    // power supply is indeed stable
+    if (!log_file.is_open())
+    {
+        LOG(WARNING) << "Cannot dmesg log file. Do you have permission?";
+        return true;
+    }
+
+    std::string line;
+    while (std::getline(log_file, line))
+    {
+        // if this lines exist, we know for sure that the battery is not stable!
+        if (line.find("soctherm: OC ALARM 0x00000001") != std::string::npos)
```

Comment:
@itsarune is this the same error message we were getting on the pi? If not, should we also check for that error message as well?

Reply:
the pi's is different but we can add the error for the Pi later on when we've confirmed support imo

Reply:
For reference, the Pi one is called "Undervoltage error"